File: quantra-backend/app/api/v1/suggestions.py
"""Quantra — Suggestions API endpoints."""
from __future__ import annotations

from fastapi import APIRouter, Depends, status
from sqlalchemy.ext.asyncio import AsyncSession

from app.dependencies import get_current_user, get_db
from app.models.user import User
from app.services.suggestion_service import get_user_suggestions, mark_suggestion_read

router = APIRouter()


@router.get("")
async def list_suggestions(user: User = Depends(get_current_user), db: AsyncSession = Depends(get_db)):
    """Today's personalized suggestions for user."""
    suggestions = await get_user_suggestions(db, user.id)
    return {
        "suggestions": [
            {"id": str(s.id), "ticker": s.ticker, "action": s.action, "confidence": float(s.confidence) if s.confidence else None,
             "reason": s.reason, "expires_at": s.expires_at.isoformat() if s.expires_at else None, "is_read": s.is_read}
            for s in suggestions
        ],
        "user_risk_profile": user.risk_profile,
    }


@router.post("/refresh", status_code=status.HTTP_202_ACCEPTED)
async def refresh_suggestions(user: User = Depends(get_current_user)):
    """Force regenerate suggestions."""
    from app.workers.tasks.generate_suggestions import run
    run.delay()
    return {"status": "queued"}


@router.patch("/{suggestion_id}/read")
async def mark_read(suggestion_id: str, user: User = Depends(get_current_user), db: AsyncSession = Depends(get_db)):
    """Mark suggestion as read."""
    success = await mark_suggestion_read(db, suggestion_id, user.id)
    if not success:
        return {"error": "Suggestion not found"}
    return {"status": "ok"}

File: quantra-frontend/src/api/suggestions.ts
import client from './client';

export interface Suggestion {
  id: string;
  ticker: string;
  name: string;
  action: 'BUY' | 'WATCH' | 'AVOID';
  ml_score: number;
  confidence: number;
  current_price: number;
  target_price: number | null;
  risk_level: 'low' | 'medium' | 'high';
  reasoning: string;
  generated_at: string;
  is_read: boolean;
}

export const suggestionsApi = {
  getSuggestions: async (): Promise<Suggestion[]> => {
    const res = await client.get('/api/v1/suggestions');
    return res.data;
  },

  refreshSuggestions: async (): Promise<{ status: string }> => {
    const res = await client.post('/api/v1/suggestions/refresh');
    return res.data;
  },

  markRead: async (id: string): Promise<void> => {
    await client.patch(`/api/v1/suggestions/${id}/read`);
  },
};
